--- src/Illuminate/Foundation/helpers.php	2020-11-25 11:58:39.156265330 -0800
+++ src/Illuminate/Foundation/helpers.php-patched	2020-11-25 12:19:26.628115587 -0800
@@ -881,7 +881,8 @@
     }
 }
 
-if (! function_exists('__')) {
+// DISABLED __ helper to avoid PKP conflict: https://github.com/pkp/pkp-lib/issues/4017
+if (false && ! function_exists('__')) {
     /**
      * Translate the given message.
      *
